### ISSUE TITLE
App appears to freeze during BT reconnect.

### ISSUE THREAD

**Issue by stieg** [autosportlabs/RaceCapture_App#920]
Issue raised on our forums: https://www.autosportlabs.org/posting.php?mode=reply&t=4734

App version 1.4.4 freezes during re-connect attempt.  From the reporter:

```
The app appears to connect but then becomes unresponsive. The only way to get it working again is to kill the app and restart. Application version is 1.4.4 on a Nexus 9 w/ Android 6.0.1 and firmware on the RCP Mk2 is 2.8.7. Also just found out that trying to copy the app log causes the app to freeze also. 
```

**Comment by ddimensia** [comment]
I did testing with a Nexus 7 running Android 5.1.1 and everything worked as expected. Another Android 6.0.1 device, Nexus 5x, had the same behavior as the Nexus 9 on 6.0.1.  The app gets to the point where it reconnects to the Bluetooth adapter, solid light on the BT module, and the read/write icons illuminate in the app but the display freezes after that and doesn't respond to input.

**Comment by ryandoherty** [comment]
Tested on RCP firmware 2.8.7, Nexus 6P/6.0.1 and app 1.3.13 and 1.4.4. I can't reproduce the issue.

Can you send us your configuration? Thanks!

**Comment by brentpicasso** [comment]
@ddimensia does this also happen on 1.3.13 (the production version).

And to clarify - the repro is::

1. Connect successfully via BT
2. power cycle RCP
3. wait for BT reconnection
4. observe hang

**Comment by ddimensia** [comment]
@brentpicasso that's the basic repo case.  Attached is my current config, I removed the telemetry deviceId as that seemed to be the only thing that probably shouldn't be shared. 

I'll try to get the production version on one of the failing devices and give it a try.

[config.rcp.txt](https://github.com/autosportlabs/RaceCapture_App/files/203155/config.rcp.txt)

**Comment by ddimensia** [comment]
I reverted the Nexus 5x to use version 1.3.13 of the app, it reconnected just fine and worked as expected.  I re-added 1.4.4 and found that it reconnects fine if telemetry streaming is turned off in the App, freezes when telemetry streaming is enabled just after the connection status text reads 'RaceCapture/Live connected'.  I tested w/ and w/o telemetry streaming on 1.3.13 and didn't have a problem.

**Comment by ryandoherty** [comment]
Awesome, that helps a lot, will test with telemetry streaming on in the app.

**Comment by ddimensia** [comment]
Not that I think there was any expectation of a change with 1.4.5 but I've confirmed the new version has the same issue with the combination of Android 6.0.1, and reconnecting with telemetry turned on in the app.

**Comment by brentpicasso** [comment]
I believe this is solved with 1.5.2, currently testing..

**Comment by ryandoherty** [comment]
99% sure this is fixed, closing.
